File: apps/ios/Barns/BarnsApp/Data/Repositories/MockProductItemRepository.swift
import Foundation

/// In-memory, local-only product items. No persistence, no network.
/// Seed data is fake, non-sensitive, and image-null compatible.
actor MockProductItemRepository: ProductItemRepository {
    private var items: [ProductItem]

    init() {
        items = [
            ProductItem(
                id: "item-wall-green-001",
                categoryId: "cat-wall-green",
                name: "Entryway green wall",
                type: .installed,
                installedOrPurchasedAt: nil,
                locationLabel: "Entryway",
                status: .active,
                careGuideIds: [],
                notes: "Installed green wall in the main entrance.",
                imageUrl: URL(string: "mock://my-greenery/my-greenery-entryway-green-wall-01"),
                updatedAt: nil
            ),
            ProductItem(
                id: "item-interior-green-001",
                categoryId: "cat-interior-green",
                name: "Reception foliage planter",
                type: .purchased,
                installedOrPurchasedAt: nil,
                locationLabel: "Reception counter",
                status: .active,
                careGuideIds: [],
                notes: "Potted foliage on the reception counter.",
                imageUrl: URL(string: "mock://my-greenery/my-greenery-reception-foliage-planter-01"),
                updatedAt: nil
            )
        ]
    }

    func productItems() async throws -> [ProductItem] {
        items
    }

    func productItem(id: ProductItem.ID) async throws -> ProductItem? {
        items.first { $0.id == id }
    }

    func addProductItem(_ item: ProductItem) async throws {
        items.append(item)
    }

    func updateProductItem(_ item: ProductItem) async throws {
        guard let index = items.firstIndex(where: { $0.id == item.id }) else { return }
        items[index] = item
    }
}

File: apps/android/app/src/main/java/com/barns/app/data/repository/MockProductItemRepository.kt
package com.barns.app.data.repository

import com.barns.app.domain.model.ProductItem
import com.barns.app.domain.model.ProductItemStatus
import com.barns.app.domain.model.ProductItemType
import com.barns.app.domain.repository.ProductItemRepository

/**
 * In-memory, local-only product items. No persistence, no network.
 * Seed data is fake, non-sensitive, and image-null compatible.
 */
class MockProductItemRepository : ProductItemRepository {
    private val lock = Any()
    private val items = mutableListOf(
        ProductItem(
            id = "item-wall-green-001",
            categoryId = "cat-wall-green",
            name = "Lobby wall greenery",
            type = ProductItemType.INSTALLED,
            installedOrPurchasedAt = null,
            locationLabel = "Entrance wall",
            status = ProductItemStatus.ACTIVE,
            careGuideIds = emptyList(),
            notes = null,
            imageUrl = null,
            updatedAt = null,
        ),
        ProductItem(
            id = "item-interior-green-001",
            categoryId = "cat-interior-green",
            name = "Desk planter",
            type = ProductItemType.PURCHASED,
            installedOrPurchasedAt = null,
            locationLabel = "Meeting room",
            status = ProductItemStatus.ACTIVE,
            careGuideIds = emptyList(),
            notes = null,
            imageUrl = null,
            updatedAt = null,
        ),
    )

    override suspend fun productItems(): List<ProductItem> =
        synchronized(lock) { items.toList() }

    override suspend fun productItem(id: String): ProductItem? =
        synchronized(lock) { items.firstOrNull { it.id == id } }

    override suspend fun addProductItem(item: ProductItem) {
        synchronized(lock) { items.add(item) }
    }

    override suspend fun updateProductItem(item: ProductItem) {
        synchronized(lock) {
            val index = items.indexOfFirst { it.id == item.id }
            if (index >= 0) items[index] = item
        }
    }
}
